File: codegen/java/examples/JJTree/src/main/java/eg4/ASTMyOtherID.java
package eg4;
import eg4.Eg4;
import eg4.Eg4Visitor;
import eg4.Node;

public
class ASTMyOtherID extends Node {
  private String name;
  public ASTMyOtherID(int id) {
    super(id);
  }

  public ASTMyOtherID(Eg4 p, int id) {
    super(p, id);
  }

  /**
   * Set the name.
   * @param n the name
   */
  public void setName(String n) {
    name = n;
  }

  /**
   * {@inheritDoc}
   * @see org.javacc.examples.jjtree.eg2.Node#toString()
   */
  public String toString() {
    return "Identifier: " + name;
  }

  /** Accept the visitor. **/
  public Object jjtAccept(Eg4Visitor visitor, Object data) {
    return visitor.visit(this, data);
  }
}

File: codegen/cpp/cpp/examples/JJTreeExamples/eg4/src/main/cpp/ASTMyOtherID.cc
/*
 * ASTMyOtherID.cpp
 *
 *  Created on: 27 mars 2014
 *      Author: FrancisANDRE
 */

#include "ASTMyOtherID.h"
#include "ParserVisitor.h"

namespace EG4 {

ASTMyOtherID::ASTMyOtherID(int id) : Node(id) {
}
ASTMyOtherID::ASTMyOtherID(Parser *p, int id) : Node(p, id){
}
ASTMyOtherID::~ASTMyOtherID() {
}

void
ASTMyOtherID::setName(JAVACC_STRING_TYPE image) {
	name = image;
}


JAVACC_STRING_TYPE
ASTMyOtherID::toString() const {
  return "Identifier: " + name;
}
/** Accept the visitor. **/
void*
ASTMyOtherID::jjtAccept(ParserVisitor *visitor, void * data) const {
  return visitor->visit(this, data);
}


} /* namespace EG4 */

